cmd/kubeadm/app/cmd/reset.go

```diff
@@ -141,6 +156,17 @@ func (r *Reset) Run(out io.Writer) error {
 	return nil
 }
 
+func resetWithDocker() {
+	dockerCheck := preflight.ServiceCheck{Service: "docker", CheckIfActive: true}
+	if _, errors := dockerCheck.Check(); len(errors) == 0 {
+		if err := exec.Command("sh", "-c", "docker ps -a --filter name=k8s_ -q | xargs -r docker rm --force --volumes").Run(); err != nil {
```

Comment:
I know this wasn't here before _sigh_, but now I really think we should have unit tests for this, as the complexity grows.
Can you please use `k8s.io/utils/exec` that you inject to the functions here, and then fake it out and test the different cases out in unit tests?

Reply:
yeah, I'll add a unit test for this function but it needs to mock out `dockerCheck` as well